src/gui/org/deidentifier/arx/gui/view/impl/common/ViewDataInput.java

```diff
@@ -199,13 +200,28 @@ public void update(final ModelEvent event) {
     
                     // Update the attribute types
                     final int index = handle.getColumnIndexOf(attr);
-                    updateHeaderImage(index, definition.getAttributeType(attr));
+                    updateHeaderImage(index, attr, definition);
     
                     // Redraw
                     table.setEnabled(true);
                     table.redraw();
                 }
             }
+        } else if (event.part == ModelPart.SELECTED_ATTRIBUTE) {

```

Comment:
I don't understand this block. Why do you need to update the header images after the user has selected a different attribute? Maybe you want to react on updated response variables? If so, you need a specific ModelEvent for this.

Reply:
And: you could implement this like it is implement for a change in AttributeType. Store the affected attribute in the even and only update the image for this specific attribute. Then, as far as I can tell, the method ViewData.updateHeaderImages is also not needed!

Reply:
My intention was to react on updated response variables by updating the icons in all relevant views. I used the wrong event. Is it correct not to introduce a new event but to use the event ATTRIBUTE_TYPE? Different icons are related to different attribute types and thus this event should have the desired effect for the case when response variables change.